symfony/framework-bundle/5.1/src/Kernel.php

```diff
@@ -15,14 +16,41 @@ protected function configureContainer(ContainerConfigurator $container): void
     {
         $container->import('../config/{packages}/*.yaml');
         $container->import('../config/{packages}/'.$this->environment.'/*.yaml');
-        $container->import('../config/{services}.yaml');
-        $container->import('../config/{services}_'.$this->environment.'.yaml');
+
+        // Put parameters here that don't need to change on each machine where the app is deployed
+        // https://symfony.com/doc/current/best_practices/configuration.html#application-related-configuration
+        //$parameters = $container->parameters();
+        //$parameters->set(...);
+
+        $services = $container->services();
+
+        // makes classes in src/ available to be used as services
+        $services
+            ->load('App\\', './*')
+            ->exclude('./{DependencyInjection,Entity,Migrations,Tests,Kernel.php}');
+
+        // controllers are imported separately to make sure services can be injected
+        // as action arguments even if you don't extend any base controller class
+        $services
+            ->load('App\\Controller\\', './Controller')
+            ->tag('controller.service_arguments');
+
+        // add more service definitions when explicit configuration is needed
+        // please note that last definitions always *replace* previous ones
+        //$services->set(...);
```

Comment:
I like this approach a lot, but I'd prefer a separate `config/services.php` file.

Reply:
That'd be way more confusing. Here, we have full context, see the type hint on the method. This makes autocompletion just work. In a separate file, you loose this context.

Reply:
what about a PHPdoc `/** @var ContainerConfigurator $container */` ?

Reply:
Make it a PR to compare :)

Reply:
It would be a pleasure, after you mark it ready :)

Reply:
I can't, the CI has a bug :)
But consider it ready, use my fork.